packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.h

```diff
@@ -141,14 +147,15 @@ class JSI_EXPORT ObjCTurboModule : public TurboModule {
       bool isSync,
       const char *methodName,
       NSInvocation *inv,
-      NSMutableArray *retainedObjectsForInvocation);
+      NSMutableArray *retainedObjectsForInvocation,
+      RCTNSDictionaryPromiseRejectBlock optionalInternalRejectBlock);
   void performVoidMethodInvocation(
       jsi::Runtime &runtime,
       const char *methodName,
       NSInvocation *inv,
       NSMutableArray *retainedObjectsForInvocation);
 
-  using PromiseInvocationBlock = void (^)(RCTPromiseResolveBlock resolveWrapper, RCTPromiseRejectBlock rejectWrapper);
+  using PromiseInvocationBlock = void (^)(RCTPromiseResolveBlock resolveWrapper, RCTPromiseRejectBlock rejectWrapper, RCTNSDictionaryPromiseRejectBlock internalRejectWrapper);
```

Comment:
Do you think it would be feasible to re-use the existing `rejectWrapper` to send the exception back to JS? Eg. you could store the nativeStack in the NSError's userInfo. (might need some changes to `RCTJSErrorFromCodeMessageAndNSError`)

Reply:
Yes, it's possible, besides allowing the params of `RCTJSErrorFromCodeMessageAndNSError` to be nullable the rest works using the `userInfo`.